### PR TITLE
feat: Tag pruner action (SOFIE-1140)

### PR DESCRIPTION
**Pull request by PeterC89**
Jira: https://nrknyemedier.atlassian.net/browse/SOFIE-1140

Create a generic action that can be used to prune tags from GitHub repositories based on age with an optional Regex filter.

Once merged the repo will need tagging with a suitable tag E.G. `v1` or `tag-pruner-v1` so that the action can be called in other repos: `nrkno/sofie-github-workflows/.github/actions/tag-pruner@v1`

An example workflow to delete tags older than 30 days that runs weekly could be:
```yaml
on:
  schedule:
    - cron: '0 0 * * 0'
name: Tidy up repo tags
jobs:
  prune:
    name: Prune tags
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
      - uses: nrkno/sofie-github-workflows/.github/actions/tag-pruner@v1
        with:
          older-than: 30
          ignore: v[0-9]+\.[0-9]+\.[0-9]+.*
```

### CODE REVIEW

File: .github/actions/tag-pruner/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+## Usage
+
+```yaml
+uses: nrkno/sofie-github-workflows/.github/actions/tag-pruner@v1
+with:
+  # Tags older than the specified number of days will be deleted (required)
+  older-than: 30
+  
+  # Regex pattern for tags that should be ignored
+  ignore: v[0-9]+\.[0-9]+\.[0-9]+.*
```
Comment (siljekristensen) [inline .github/actions/tag-pruner/README.md:14]: If you have several patterns that should be ignored, how should they be added?

**Review by siljekristensen — COMMENTED** [review]
Have you tested using this action in another repo?

File: .github/actions/tag-pruner/README.md (hunk 1 above)
Comment (PeterC89) [reply .github/actions/tag-pruner/README.md:14]: Currently by writing a very complex regex pattern that covers all of them?
Not sure what happens if you try to pass an array as an input to an action, the typings suggest all inputs are converted to string but I'll try it and see what happens!

**Comment by PeterC89** [comment]
> Have you tested using this action in another repo?

Yes, I built this in one repo and called it from another, seemed to work!

File: .github/actions/tag-pruner/README.md (hunk 1 above)
Comment (PeterC89) [reply .github/actions/tag-pruner/README.md:14]: So it literally takes the value of an environment variable called `INPUT_IGNORE` and passes that back as a string: https://github.com/actions/toolkit/blob/03d6c2479c60c5873d1959fe1f167cc57dccccfc/packages/core/src/core.ts#L126

I guess realistically the options are either to have multiple regex input options, E.G. ignore-1, ignore-2 etc.
Or we could say that a comma is used as a delimiter for multiple patterns so you could write something like: `pattern1,pattern2` although that feels a little messy and means we can't have commas in the regex? (Probably not a big problem)
Or to write a complex pattern?

Comment (siljekristensen) [reply .github/actions/tag-pruner/README.md:14]: I guess one complex pattern works, I just like regex to be as simple as possible.

File: .github/actions/tag-pruner/action.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: 'Sofie Git repository tag tidier'
+description: 'Provide a description here'
```
Comment (siljekristensen) [inline .github/actions/tag-pruner/action.yml]: Perhaps add a description?

**Review by siljekristensen — APPROVED** [review]

File: .github/actions/tag-pruner/action.yml (hunk 2 above)
Comment (PeterC89) [reply .github/actions/tag-pruner/action.yml]: Whoops, done!

**Pull request merged**